lib/ansible/galaxy/collection.py

```diff
@@ -572,9 +572,12 @@ def _get_galaxy_yml(b_galaxy_yml_path):
     return galaxy_yml
 
 
-def _build_files_manifest(b_collection_path):
-    b_ignore_files = frozenset([b'*.pyc', b'*.retry'])
-    b_ignore_dirs = frozenset([b'CVS', b'.bzr', b'.hg', b'.git', b'.svn', b'__pycache__', b'.tox'])
+def _build_files_manifest(b_collection_path, namespace, name):
+    # Contains tuple of (b_filename, only root) where 'only root' means to only ignore the file in the root dir
+    b_ignore_files = frozenset([(b'*.pyc', False), (b'*.retry', False),
+                                (to_bytes('{0}-{1}-*.tar.gz'.format(namespace, name)), True)])
+    b_ignore_dirs = frozenset([(b'CVS', False), (b'.bzr', False), (b'.hg', False), (b'.git', False), (b'.svn', False),
+                               (b'__pycache__', False), (b'.tox', False)])
```

Comment:
not for this PR, but we might want a 'configurable ignore facility' ala gitignore in the future